exporters/otlp/recordable.cc

```diff
@@ -5,6 +5,30 @@ namespace exporter
 {
 namespace otlp
 {
+void Recordable::SetIds(trace::TraceId trace_id,
+                        trace::SpanId span_id,
+                        trace::SpanId parent_span_id) noexcept
+{
+  const uint8_t* trace_id_data = trace_id.Id().data();
+  const uint8_t* span_id_data = span_id.Id().data();
+  const uint8_t* parent_span_id_data = parent_span_id.Id().data();
+
+  const char* trace_id_buff = (char*)trace_id_data;
+  const char* span_id_buff = (char*)span_id_data;
+  const char* parent_span_id_buff = (char*)parent_span_id_data;
+
+  span_.set_trace_id(trace_id_buff);
```

Comment:
I think there is a bug here that together with a bug in test makes test pass.
If trace_id_buff contains any null character, the string will only be copied up to that first null character.
It is necessary to specify size as a second argument in setting trace_id (and other ids below)

Also, I would probably get rid of extra variables for data and buff and convert it as

span_.set_trace_id(reinterpret_cast<const char*>(trace_id.Id().data()), trace::TraceId::kSize);